docs/rules/no-children-prop.md

```diff
@@ -34,3 +34,29 @@ The following patterns are **not** considered warnings:
 React.createElement("div", {}, 'Children')
 React.createElement("div", 'Child 1', 'Child 2')
 ```
+
+## Rule Options
+
+```js
+"react/no-children-prop": [<enabled>, {
+  "allowFunctions": <boolean> || false
+}]
+```
+
+### `allowFunctions`
+
+When `true`, passing a function in the children prop is preferred.
```

Comment:
let's say this instead: when true, and passing a function as `children`, it must be in prop position and not child position.